pdl-compiler/src/backends/rust/parser.rs

```diff
@@ -513,6 +533,108 @@ impl<'a> FieldParser<'a> {
                     }
                 });
             }
+            (ElementWidth::Dynamic(element_size_field), ArrayShape::Static(count)) => {
+                // The element width is known, and the array element
+                // count is known statically.
+                let array_size = if *count == 1 {
+                    quote!(#element_size_field)
+                } else {
+                    quote!(#count * #element_size_field)
+                };
+
+                self.check_size(&span, &array_size);
+
+                let parse_element =
+                    self.parse_array_element(&format_ident!("chunk"), width, type_id, decl);
+
+                self.code.push(quote! {
+                    // TODO: use
+                    // https://doc.rust-lang.org/std/array/fn.try_from_fn.html
+                    // when stabilized.
+                    let #id = #span.chunks(#element_size_field)
+                        .take(#count)
+                        .map(|mut chunk| #parse_element.and_then(|value| {
+                            if chunk.is_empty() {
+                                Ok(value)
+                            } else {
+                                Err(DecodeError::TrailingBytesInArray {
+                                    obj: #packet_name,
+                                    field: #field_name,
+                                })
+                            }
+                         }))
+                        .collect::<Result<Vec<_>, DecodeError>>()?;
+                    #span = &#span[#array_size..];
+                    let #id = #id
+                        .try_into()
+                        .map_err(|_| DecodeError::InvalidPacketError)?;
```

Comment:
I think it is a valid case of panic here.

Reply:
It's not a panic for the other cases that uses `ArrayShape::Static`, I'm fine doing it, but it should be done in a follow up to all `ArrayShape::Static`